### DISCUSSION TITLE
are the inputs always assumed to be tokens?

### DISCUSSION THREAD

**Discussion by kashif** [facebookresearch/xformers#126] [Q&A] [answered]
Hi, quick question,

does the API assume the inputs are always tokens or can one give a set of vector inputs like in the torch transformer's encoder decoder modules?

Thanks!

**Comment by kashif** [comment, marked as answer]
ok I figured it out I believe, I can just set the `position_encoding_config` to `None`

**Comment by blefaudeux** [comment]
oh sorry, I just saw this too late. You figured it out, there's not specific assumption, it all depends on the build blocks being present. If you're not passing tokens then probably that you would add sine encodings though ? In any case feel free to compose models the way you want, the construction with a config presented in the examples is just that, an example, I think that it can be useful for something like architecture search but maybe not for all usages. Thanks for the question @kashif !
